samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -184,7 +184,9 @@ static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
 	}
 
 	/* We're only interested in connectable events */
-	if (type != BT_GAP_ADV_TYPE_ADV_IND && type != BT_GAP_ADV_TYPE_ADV_DIRECT_IND) {
+	if (type != BT_GAP_ADV_TYPE_ADV_IND &&
+	    type != BT_GAP_ADV_TYPE_ADV_DIRECT_IND &&
+	    type != BT_GAP_ADV_TYPE_EXT_ADV) {
```

Comment:
It's an OK fix for now, but the proper fix is to use the new callback structure for scanning, and check the `info->adv_props`. 

See e.g. the check from csis_client.c (shell):

```c
static void csis_client_scan_recv(const struct bt_le_scan_recv_info *info,
				  struct net_buf_simple *ad)
{
	/* We're only interested in connectable events */
	if (info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) {
		if (cur_inst != NULL) {
			bt_data_parse(ad, csis_found, (void *)info->addr);
		}
	}
}
```

Reply:
I agree this suggestion is better, but as it requires API changes to the stack I do not think it should be solved in a PR to fix a sample application, but rather have a new issue requesting this enhancement of adding the `bt_le_scan_recv_info ` to the call-back.